lib/fastly_nsq/listener.rb

```diff
@@ -0,0 +1,59 @@
+require 'fastly_nsq/message'
+
+module FastlyNsq
+  class Listener
+    def self.listen_to(**args)
+      new(**args).go
+    end
+
+    def initialize(topic:, processor:, channel: nil, consumer: nil, **options)
```

Comment:
It's hard to see here with the rename, but keep in mind this has [not really changed relative to current master](https://github.com/fastly/fastly_nsq/blob/f0da289828b02c0fd9683ddb1832257e58b3b7aa/lib/fastly_nsq/message_queue/listener.rb).

Originally, `Listener` required `topic` & `channel` and managed `consumer` internally. (`topic` is still passed to `processor`, but currently all of our clients ignore it, so it, too, could be made optional.) 

Personally, I'd eventually like to see this drop `channel` *and* `topic, and initialize `Listener` with a `Consumer`. The main challenge here is that `Listener`'s specs indicate that it sort of wants to wrap & manage `Consumer` (lazily instantiating and eventually terminating). 

I'd like to move this up into something like `FastlyNsq.listen_to`, let that method coordinate `Consumer` and simplify `Listener` a great deal) but I felt this PR was already sprawling in scope, and this change wasn't necessary even in the (currently enlarged) scope.